pkg/fqdn/proxy/proxy.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2016-2020 Authors of Cilium
+
+package proxy
+
+import (
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/fqdn/restore"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+type FQDNProxy interface {
+	GetRules(uint16) (restore.DNSRules, error)
```

Comment:
Why does this need to return also an error given that both the `DNSProxy` and the `MockFQDNProxy` implementations cannot fail?

Reply:
FWIW the DNSProxy implementation does have error conditions, we just don't hand them back up the stack at the moment. Ultimately I think that the current error condition should not be blocking the implementation of DNS policy so it doesn't make a huge difference where we handle that error, but technically it could be up to the parent functions to handle those errors appropriately (for instance, by outputting them to the endpoint-specific log rather than spamming the global log like the current implementation does).